enhancements/declarative-index-config.md

```diff
@@ -1242,11 +1119,8 @@ Additionally, `prune`, `prune-stranded` and `rm` sub-commands under the `opm ind
 
 ## Migration Plan
 
-The `opm` command will automatically migrate an sqlite-based database to a declarative config-based database according to the following rules:
-1. Existing `opm index` commands will support images containing the deprecated sqlite representation or the declarative config-based tar or directory representation. All `opm index` subcommands will automatically migrate the input database to the declarative config directory representation, and then perform the desired underlying registry action against the declarative config representation.
-2. Existing `opm registry` commands will support input databases with the deprecated sqlite representation or the declarative config-based tar or directory representation. If the input database is a file (e.g. sqlite or declarative config tar), `opm registry` commands that modify the database will always output the declarative config tar format. This ensures that existing clients of `opm registry` who have not explicitly migrated can continue treating the database as an opaque file. If the input database is a declarative config directory, `opm registry` preserves the fact that it is a directory. `opm registry serve` does not have any permanent filesystem side effects.
-3. For existing commands, no new flags are being introduced to distinguish sqlite input from declarative config input. All input database formats can be provided via the `--database` flag.
-
+Migration is opt-in. Users with existing index images and database files can continue using `opm` in their current workflows
```

Comment:
Should we make it clear here that migration is opt-in but will remain so only for the period of time serving with database remains deprecated? Users must migrate within that period.

Reply:
This brings up an interesting point which is that we should call out a deprecation notice for the sqlite database (and as a result all of the old commands). I'm also imagining that in whatever release we actually remove those old commands, the release before it *should* do some automatic migration so nobody gets stuck

Reply:
I thought about touching on that, but it is still not clear to me from the discussions how we intend to help folks do the explicit migration. I think the options are:
1. Rebuild via a bunch of `opm add` commands.
2. We would provide an explicit migration command.

Either way, I think the way this is worded now is "good enough". We don't say when support for sqlite will be removed anywhere, and I think the fact that we're saying something is deprecated has an implicit suggestion that a migration will eventually be necessary.

Reply:
I DO think we should expand on this topic in a follow-up though.

Reply:
That sounds reasonable. Since we don't have time to tackle all of this for 4.8, 4.9 is where we should deprecate all of the commands we want to, provide an explicit migration path, i.e `opm registry generate-configs`, and then 4.10 we can automatically migrate any remaining databases to configs. Provides us with a healthy amount of time to tackle mirroring with oc too.    
By 4.11 the sqllite database will be completely removed.

Reply:
As the adoption/migration actions continue to be revised, i'd ask that you revisit the "pipeline decalarative config adoption" guide and

1) update it to reflect the steps the pipeline will need to take
2) ensure we are reconfirming w/ the pipeline team that they are ok w/ those steps/requirements.

https://docs.google.com/document/d/15nasm_NQBhAdl5FzY4ZO0SULP8CrP2XB0vl4fq9xSnc/edit?ts=603015b4#heading=h.82pzi4nih86k